pkg/api/datasources.go

```diff
@@ -103,14 +104,61 @@ func DeleteDataSourceById(c *models.ReqContext) Response {
 	return Success("Data source deleted")
 }
 
+// GET /api/datasources/uid/:uid
+func GetDataSourceByUID(c *models.ReqContext) Response {
+	query := models.GetDataSourceQuery{
+		Uid:   c.Params(":uid"),
+		OrgId: c.OrgId,
+	}
+
+	if err := bus.Dispatch(&query); err != nil {
+		if errors.Is(err, models.ErrDataSourceNotFound) {
+			return Error(404, "Data source not found", nil)
+		}
+		return Error(500, "Failed to query datasources", err)
+	}
+
+	ds := query.Result
+	dtos := convertModelToDtos(ds)
+
+	return JSON(200, &dtos)
+}
+
+// DELETE /api/datasources/uid/:uid
+func DeleteDataSourceByUID(c *models.ReqContext) Response {
+	uid := c.Params(":uid")
+
+	if uid == "" {
+		return Error(400, "Missing datasource uid", nil)
+	}
+
+	ds, err := getRawDataSourceByUID(uid, c.OrgId)
+	if err != nil {
+		return Error(400, "Failed to delete datasource", nil)
+	}
+
+	if ds.ReadOnly {
+		return Error(403, "Cannot delete read-only data source", nil)
+	}
+
+	cmd := &models.DeleteDataSourceByUIDCommand{Uid: uid, OrgId: c.OrgId}
```

Comment:
Nit. Could potentially use one `DeleteDataSourceCommand` similar to `GetDataSourceQuery` for the different kind of deletes to limit some code duplication.

Reply:
yea, I think that makes sense as well. less confusion on what command to use